tasks_11/database_class_3.py

```diff
@@ -0,0 +1,122 @@
+"""
+Задача 3
+Реализовать программу с базой учащихся группы (данные хранятся в файле).
+Записи по учащемуся должны быть представлены отдельным классом с полями:
+имя, фамилия, пол, возраст. Программа должна предусматривать поиск по
+одному или нескольким полям базы. Результат выводить в удобочитаемом виде
+с порядковым номером записи. Скрипт программы должен принимать параметр,
+который определяет формат хранения и реализации БД: в текстовом файле с
+определенной структурой; в файле json.
+"""
+
+import json
+import sys
+import os
+
+
+class BaseDB:
+    db = []
+
+    def __init__(self, filename):
+        self.filename = filename
+        self.db = self.load(filename)
+
+    def load(self, filename):
+        raise NotImplementedError
+
+    def find(self, search):
+        criteria = set(search)
+        students_list = []
+
+        for id, student in enumerate(self.db, 1):
```

Comment:
try to avoid of `id` using due to `id()` is a builtin function.
`student_id` might be more suitable.